### PR TITLE
Externalize Standard Job Timeout

### PR DESCRIPTION
**Pull request by dealako**
* Jobs can now override the job standard timeout

JIRA: RE-2307

### CODE REVIEW

File: pipeline_steps/common.groovy [hunk 1]
```diff
@@ -1240,19 +1240,19 @@ void internal_slave(Closure body){
 }
 
 void standard_job_slave(String slave_type, Closure body){
-  timeout(time: 8, unit: 'HOURS'){
+  timeout(time: env.JOB_TIMEOUT_HR, unit: 'HOURS'){
```
Comment (hughsaunders) [inline pipeline_steps/common.groovy]: This isn't really a job timeout, this is a slave timeout. A job may use multiple slaves, in parallel or serial, and its not unreasonable for the job as a whole to take longer than any one slave is used for. 

I guess what I'm saying is that if these are going to be tweak-able they should probably be tweak-able independently.

Both should have a default, so most users will only need to set `JOB_TIMEOUT` but `AGENT_TIMEOUT` could be useful in some situations.

File: rpc_jobs/unit/job_timeout.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+- job:
+    name: RE-unit-test-job-timeout
```
Comment (hughsaunders) [inline rpc_jobs/unit/job_timeout.yml]: I think this should be a lint check.

File: pipeline_steps/common.groovy (hunk 1 above)
Comment (dealako) [reply pipeline_steps/common.groovy]: Refactored to accomodate additional parameter.

File: rpc_jobs/unit/job_timeout.yml (hunk 2 above)
Comment (dealako) [reply rpc_jobs/unit/job_timeout.yml]: Removed - moved logic to jjb_lint.py

**Comment by dealako** [comment]
Closing this PR as we need these changes to be introduced as two separate commits to allow our existing pre-merge checks to pass.

**Pull request closed without merging**